torch/csrc/jit/python/init.cpp

```diff
@@ -1366,13 +1366,14 @@ void initJITBindings(PyObject* module) {
           "fallback", [](GraphExecutorState& s) { return s.fallback; });
 
   py::class_<PyTorchStreamWriter>(m, "PyTorchFileWriter")
-      .def(py::init<std::string>())
+      .def(py::init<std::string>(), py::call_guard<py::gil_scoped_release>())
       .def(py::init([](const py::object& buffer) {
         auto writer_func = [=](const void* data, size_t size) {
           // Writing an empty file is a noop
           if (size == 0) {
             return size;
           }
+          py::gil_scoped_acquire g;
```

Comment:
If I understand correctly, this is a callback that will get called by `write_record` to write to a python object. Since `write_record` released the gil, it's acquired again here.